service/history/historyBuilder.go

```diff
@@ -591,14 +598,22 @@ func (b *historyBuilder) newActivityTaskScheduledEvent(decisionTaskCompletedEven
 	return historyEvent
 }
 
-func (b *historyBuilder) newActivityTaskStartedEvent(scheduledEventID int64, attempt int32, requestID string,
-	identity string) *workflow.HistoryEvent {
+func (b *historyBuilder) newActivityTaskStartedEvent(
+	scheduledEventID int64,
+	attempt int32,
+	requestID string,
+	identity string,
+	lastFailureReason string,
+	lastFailureDetails []byte,
+) *workflow.HistoryEvent {
 	historyEvent := b.msBuilder.CreateNewHistoryEvent(workflow.EventTypeActivityTaskStarted)
 	attributes := &workflow.ActivityTaskStartedEventAttributes{}
 	attributes.ScheduledEventId = common.Int64Ptr(scheduledEventID)
 	attributes.Attempt = common.Int32Ptr(attempt)
 	attributes.Identity = common.StringPtr(identity)
 	attributes.RequestId = common.StringPtr(requestID)
+	attributes.LastFailureReason = common.StringPtr(lastFailureReason)
+	attributes.LastFailureDetails = lastFailureDetails
```

Comment:
is it possible to update any existing test to validate that these values are now set when activity fails ?

Reply:
+1